app/src/pages/playground/PlaygroundExperimentRunDetailsDialog.tsx

```diff
@@ -22,73 +36,171 @@ export function PlaygroundExperimentRunDetailsDialog({
-      exampleId: example?.id,
-      input: JSON.stringify(revision?.input, null, 2),
-      referenceOutput: JSON.stringify(revision?.output, null, 2),
-      runOutput:
-        run.output !== null ? JSON.stringify(run.output, null, 2) : null,
-    };
-  }, [data]);
-  const { exampleId, input, referenceOutput, runOutput } = run;
+  const run = data.run;
+  const exampleId = run.example?.id;
+  const revision = run.example?.revision;
+  const input = revision?.input;
+  const referenceOutput = revision?.output;
   return (
-    <Dialog size="XL" title={`Experiment Run for Example: ${exampleId}`}>
-      <Flex direction="row" justifyContent="center">
-        <View
-          width="900px"
-          paddingStart="auto"
-          paddingEnd="auto"
-          paddingTop="size-200"
-          paddingBottom="size-200"
-        >
-          <Flex direction="column" gap="size-200">
-            <Card
-              title="Input"
-              {...defaultCardProps}
-              extra={<CopyToClipboardButton text={input} />}
+    <Dialog
+      title={`Experiment Run for Example: ${exampleId}`}
+      size="fullscreen"
+    >
+      <PanelGroup direction="vertical" autoSaveId="example-compare-panel-group">
+        <Panel defaultSize={100}>
+          <div
+            css={css`
+              overflow-y: auto;
+              height: 100%;
+            `}
+          >
+            <View overflow="hidden" padding="size-200">
+              <Flex direction="row" gap="size-200" flex="1 1 auto">
+                <View width="50%">
+                  <Card
+                    title="Input"
+                    {...defaultCardProps}
+                    bodyStyle={{
+                      padding: 0,
+                      maxHeight: "300px",
+                      overflowY: "auto",
+                    }}
+                    extra={
+                      <CopyToClipboardButton text={JSON.stringify(input)} />
+                    }
+                  >
+                    <JSONBlock value={JSON.stringify(input, null, 2)} />
+                  </Card>
+                </View>
+                <View width="50%">
+                  <Card
+                    title="Reference Output"
+                    {...defaultCardProps}
+                    extra={
+                      <CopyToClipboardButton
+                        text={JSON.stringify(referenceOutput)}
+                      />
+                    }
+                    bodyStyle={{
+                      padding: 0,
+                      maxHeight: "300px",
+                      overflowY: "auto",
+                    }}
+                  >
+                    <JSONBlock
+                      value={JSON.stringify(referenceOutput, null, 2)}
+                    />
+                  </Card>
+                </View>
+              </Flex>
+            </View>
+          </div>
+        </Panel>
+        <PanelResizeHandle css={resizeHandleCSS} />
+        <Panel defaultSize={200}>
```

Comment:
These sizes are a percentage of the total PanelGroup component and should not exceed 100 nor represent pixels

https://github.com/bvaughn/react-resizable-panels/tree/main/packages/react-resizable-panels#panel